test/provisioner/common.go

```diff
@@ -0,0 +1,62 @@
+package provisioner
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+type patchLabel struct {
+	Op    string            `json:"op"`
+	Path  string            `json:"path"`
+	Value map[string]string `json:"value"`
+}
+
+func AddNodeRoleWorkerLabel(ctx context.Context, clusterName string, cfg *envconf.Config) error {
```

Comment:
did you try to extract the cluster name from the Config and/or Client objects? If it is possible, it would save us one argument to this function :)

Reply:
While talking about this function, as it will have public access...could you please add documentation?